### PR TITLE
Docs: migration to 25 guide.

### PR DESCRIPTION
**Pull request by godai78**
Type: Docs: Adding the Migration to v25.0.0 guide to the docs.

Closes: cksource/collaboration-features#3877

Necessary follow-up - link the guide in the CF changelog here: https://github.com/cksource/collaboration-features/blob/master/CHANGELOG.md#changes-in-the-annotations-api

### CODE REVIEW

**Comment by godai78** [comment]
@ckeditor/qa-team - guys, the same story, same error. Have a look at that, please?

**Comment by godai78** [comment]
CI'd properly, looks like good to go.

File: docs/builds/guides/migration/migration-to-25.0.0.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+In place where the [`AnnotationView`](https://ckeditor.com/docs/ckeditor5/latest/api/module_comments_annotations_view_annotationview-AnnotationView.html) had been used, the new [`Annotations`](https://ckeditor.com/docs/ckeditor5/latest/api/module_comments_annotations_annotations-Annotations.html)  class instance is used and it exposes mainly the following properties:
+* [`target`](https://ckeditor.com/docs/ckeditor5/latest/api/module_comments_annotations_annotation-Annotation.html#member-target) - The target element to which the annotation should be bound. It can be a [`Rect`](https://ckeditor.com/docs/ckeditor5/latest/api/module_utils_dom_rect-Rect.html) instance or an HTML Element.
+* [`view`](https://ckeditor.com/docs/ckeditor5/latest/api/module_comments_annotations_annotation-Annotation.html#member-view) - The [`AnnotationView`](https://ckeditor.com/docs/ckeditor5/latest/api/module_comments_annotations_view_annotationview-AnnotationView.html) instance.
+* [`type`](https://ckeditor.com/docs/ckeditor5/latest/api/module_comments_annotations_annotation-Annotation.html#member-type) - The type of the annotation.
```
Comment (AnnaTomanek) [inline docs/builds/guides/migration/migration-to-25.0.0.md:35]:
```suggestion
* [`type`](https://ckeditor.com/docs/ckeditor5/latest/api/module_comments_annotations_annotation-Annotation.html#member-type) &ndash; The type of the annotation.
```

File: docs/builds/guides/migration/migration-to-25.0.0.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+* [`isActive`](https://ckeditor.com/docs/ckeditor5/latest/api/module_comments_annotations_annotation-Annotation.html#member-isActive) - A property that determines if the annotation view and annotation inner views are active or not. It should be changed by the annotation UI when it changes the active annotation.
```
Comment (AnnaTomanek) [inline docs/builds/guides/migration/migration-to-25.0.0.md:36]:
```suggestion
* [`isActive`](https://ckeditor.com/docs/ckeditor5/latest/api/module_comments_annotations_annotation-Annotation.html#member-isActive) &ndash; A property that determines if the annotation view and annotation inner views are active or not. It should be changed by the annotation UI when it changes the active annotation.
```

File: docs/builds/guides/migration/migration-to-25.0.0.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+The [`Sidebar`](https://ckeditor.com/docs/ckeditor5/latest/api/module_comments_annotations_sidebar-Sidebar.html) class stopped operating on the global collection of annotations. From now on, the Annotation UI that is responsible for displaying the sidebar should provide the sidebar with its collection of annotations and inform about possible actions with the new [`rearrange()`](https://ckeditor.com/docs/ckeditor5/latest/api/module_comments_annotations_sidebar-Sidebar.html#function-rearrange) and [`refresh()`](https://ckeditor.com/docs/ckeditor5/latest/api/module_comments_annotations_sidebar-Sidebar.html#function-refresh) methods.
```
Comment (AnnaTomanek) [inline docs/builds/guides/migration/migration-to-25.0.0.md:38]:
```suggestion
The [`Sidebar`](https://ckeditor.com/docs/ckeditor5/latest/api/module_comments_annotations_sidebar-Sidebar.html) class stopped operating on the global collection of annotations. From now on, the annotation UI that is responsible for displaying the sidebar should provide the sidebar with its collection of annotations and inform about possible actions with the new [`rearrange()`](https://ckeditor.com/docs/ckeditor5/latest/api/module_comments_annotations_sidebar-Sidebar.html#function-rearrange) and [`refresh()`](https://ckeditor.com/docs/ckeditor5/latest/api/module_comments_annotations_sidebar-Sidebar.html#function-refresh) methods.
```

File: docs/builds/guides/migration/migration-to-25.0.0.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+		}
+
+		this.activeAnnotation = annotation;
+	}
+
+	attach( annotations ) {
+		// The code responsible for displaying annotations
+		// based on the annotation collection passed to this UI.
+
+		// The code responsible for setting an active and non-active annotation based on the annotation focus changes.
```
Comment (AnnaTomanek) [inline docs/builds/guides/migration/migration-to-25.0.0.md:173]:
```suggestion
		// The code responsible for setting an active and non-active annotation
		// based on the annotation focus changes.
```

File: docs/builds/guides/migration/migration-to-25.0.0.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+		} );
+
+		this.listenTo( annotations, 'blur', () => {
+			this.setActiveAnnotation( null );
+		} );
+
+		// The code responsible for the integration with editor annotation markers,
+		// editor events and the editor selection changes.
+		// The requirement of this integration might change in the future.
+		// // If the plugin was initialized as an editor plugin, the integration should look like the following:
```
Comment (AnnaTomanek) [inline docs/builds/guides/migration/migration-to-25.0.0.md:185]:
```suggestion
		// If the plugin was initialized as an editor plugin, the integration
		// should look like the following:
```

File: docs/builds/guides/migration/migration-to-26.0.0.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
 ---
 category: builds-migration
```
Comment (AnnaTomanek) [inline docs/builds/guides/migration/migration-to-26.0.0.md:2]:
```suggestion
category: builds-migration
menu-title: Migration to v26.0.0
```

**Review by AnnaTomanek — CHANGES_REQUESTED** [review]

**Review by AnnaTomanek — APPROVED** [review]

**Pull request merged**